findflows antenna detection initial commit

iavssw opened the pull request:
This is the initial implementation of the antenna detection solution in the findflows pass.

stephenneuendorffer commented on lib/AIEFindFlows.cpp:
```diff
@@ -133,20 +228,115 @@ class ConnectivityAnalysis {
       if (!nextConnection.hasValue())
         continue;
-
+      PortConnection sour = std::make_pair(switchOp, nextPort);
+      PacketConnection dest = std::make_pair(nextConnection.getValue(), newMaskValue);
```
generally prefer full words "source" and "destination"

iavssw commented:
Thanks for the feedback. I’ll try to improve the code based on your feedback.
Also, I’ve tracked down the reason for my email not working.  It was due to the mail being forwarded to the amd account. I’ve now logged in using amd credentials and will be able to respond to you via that email.

Greg
________________________________
From: Stephen Neuendorffer ***@***.***>
Sent: Thursday, June 2, 2022 12:03:26 PM
To: Xilinx/mlir-aie ***@***.***>
Cc: Jun, Gregory ***@***.***>; Author ***@***.***>
Subject: Re: [Xilinx/mlir-aie] findflows antenna detection initial commit (PR #107)


@stephenneuendorffer commented on this pull request.

________________________________

In lib/AIEFindFlows.cpp<https://urldefense.com/v3/__https://github.com/Xilinx/mlir-aie/pull/107*discussion_r888173602__;Iw!!DZ3fjg!4R5wAJj99YPI16Q1h3iVHn8NEXlVMg15kzmHQ-em7MhNmR3DhhDtManlyqcA3nvWcbsKUZSGSwUUJ8MDdWMbmN2gv5s$>:

>  typedef std::pair<Operation *, Port> PortConnection;
 typedef std::pair<Port, MaskValue> PortMaskValue;
 typedef std::pair<PortConnection, MaskValue> PacketConnection;
+typedef std::pair<PortConnection, PacketConnection> Source_PacketConnection;


I think we're past the point where more cascaded std::pair's is a good thing here. Could you simplify this?

________________________________

In lib/AIEFindFlows.cpp<https://urldefense.com/v3/__https://github.com/Xilinx/mlir-aie/pull/107*discussion_r888186745__;Iw!!DZ3fjg!4R5wAJj99YPI16Q1h3iVHn8NEXlVMg15kzmHQ-em7MhNmR3DhhDtManlyqcA3nvWcbsKUZSGSwUUJ8MDdWMb9ROtiWo$>:

> +      PortConnection sour = std::make_pair(switchOp, nextPort);
+      PacketConnection dest = std::make_pair(nextConnection.getValue(), newMaskValue);


generally prefer full words "source" and "destination"

________________________________

In lib/AIEFindFlows.cpp<https://urldefense.com/v3/__https://github.com/Xilinx/mlir-aie/pull/107*discussion_r888187892__;Iw!!DZ3fjg!4R5wAJj99YPI16Q1h3iVHn8NEXlVMg15kzmHQ-em7MhNmR3DhhDtManlyqcA3nvWcbsKUZSGSwUUJ8MDdWMbjSxNKjA$>:

>  typedef std::pair<Operation *, Port> PortConnection;
 typedef std::pair<Port, MaskValue> PortMaskValue;
 typedef std::pair<PortConnection, MaskValue> PacketConnection;
+typedef std::pair<PortConnection, PacketConnection> Source_PacketConnection;


It seems like the 'Source_PacketConnection' has all the information you need to represent the graph. Maybe it could just be a shared_ptr<> to the source, which would remove the need for the separate graph data structure?

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https://github.com/Xilinx/mlir-aie/pull/107*pullrequestreview-993857867__;Iw!!DZ3fjg!4R5wAJj99YPI16Q1h3iVHn8NEXlVMg15kzmHQ-em7MhNmR3DhhDtManlyqcA3nvWcbsKUZSGSwUUJ8MDdWMb8CImeHU$>, or unsubscribe<https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AT7SR66L4WWDDFWVXMS6XZLVNDSN5ANCNFSM5XUEXMTQ__;!!DZ3fjg!4R5wAJj99YPI16Q1h3iVHn8NEXlVMg15kzmHQ-em7MhNmR3DhhDtManlyqcA3nvWcbsKUZSGSwUUJ8MDdWMbs-YQaDA$>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

iavssw commented:
Version 2 of the antenna detection feature update. Removed bugs. Simplified the code to minimize redundant operations. Now using the MLIR Diagnostic Infrastructure to output messages.
